R/expr-string.R

```diff
@@ -0,0 +1,1138 @@
+#' @param exact If `TRUE` (default), require an exact format match. If `FALSE`, allow the format to match
+#' anywhere in the target string. Note that using `exact = FALSE` introduces a performance
+#' penalty - cleaning your data beforehand will almost certainly be more performant.
+#' @inherit as_polars_expr return
+#' @seealso
+#' - [`<Expr>$str$strptime()`][expr_str_strptime]
+#' @examples
+#' s <- as_polars_series(c("2020-01-01 01:00Z", "2020-01-01 02:00Z"))
+#'
+#' s$str$to_datetime("%Y-%m-%d %H:%M%#z")
+#' s$str$to_datetime(time_unit = "ms")
+expr_str_to_datetime <- function(
+    format = NULL,
+    ...,
+    time_unit = NULL,
+    time_zone = NULL,
+    strict = TRUE,
+    exact = TRUE,
+    cache = TRUE,
+    ambiguous = "raise") {
+  wrap({
+    check_dots_empty0(...)
+    self$`_rexpr`$str_to_datetime(
+      self, format, time_unit, time_zone, strict, exact, cache, ambiguous
+    )
+  })
+}
+
+#' Get the number of bytes in strings
+#' @description
+#' Get length of the strings as UInt32 (as number of bytes). Use `$str$len_chars()`
+#' to get the number of characters.
+#'
+#' @details
+#' If you know that you are working with ASCII text, `lengths` will be
+#' equivalent, and faster (returns length in terms of the number of bytes).
+#' @inherit as_polars_expr return
+#' @examples
+#' pl$DataFrame(
+#'   s = c("Café", NA, "345", "æøå")
+#' )$select(
+#'   pl$col("s"),
+#'   pl$col("s")$str$len_bytes()$alias("lengths"),
+#'   pl$col("s")$str$len_chars()$alias("n_chars")
+#' )
+expr_str_len_bytes <- function() {
+  self$`_rexpr`$str_len_bytes() |>
+    wrap()
+}
+
+#' Get the number of characters in strings
+#' @description
+#' Get length of the strings as UInt32 (as number of characters). Use
+#' `$str$len_bytes()` to get the number of bytes.
+#'
+#' @inherit expr_str_len_bytes examples details return
+expr_str_len_chars <- function() {
+  self$`_rexpr`$str_len_chars() |>
+    wrap()
+}
+
+#' Vertically concatenate the string values in the column to a single string value.
+#'
+#' @param delimiter The delimiter to insert between consecutive string values.
+#' @inheritParams rlang::args_dots_empty
+#' @param ignore_nulls Ignore null values (default). If `FALSE`, null values will be
+#' propagated: if the column contains any null values, the output is null.
+#' @inherit as_polars_expr return
+#' @examples
+#' # concatenate a Series of strings to a single string
+#' df <- pl$DataFrame(foo = c(1, NA, 2))
+#'
+#' df$select(pl$col("foo")$str$join("-"))
+#'
+#' df$select(pl$col("foo")$str$join("-", ignore_nulls = FALSE))
+expr_str_join <- function(
+    delimiter = "",
+    ...,
+    ignore_nulls = TRUE) {
+  wrap({
+    check_dots_empty0(...)
+    self$`_rexpr`$str_join(delimiter, ignore_nulls)
+  })
+}
+
+expr_str_concat <- function(
+    delimiter = "",
+    ...,
+    ignore_nulls = TRUE) {
+  lifecycle::deprecate_warn("0.18.0", what = I("$str$concat()"), with = I("$str$join()"))
```

Comment:
This package does not have lifecycle as a dependency. Use this instead.
https://github.com/eitsupi/neo-r-polars/blob/b90688621f7d6c4e35e34574421bc63122c58f35/R/import-standalone-lifecycle.R#L129